modules/tinymce/src/core/main/ts/api/commands/LinkCommands.ts

```diff
@@ -8,6 +8,14 @@ export const registerCommands = (editor: Editor): void => {
     const anchor = editor.dom.getParent(editor.selection.getNode(), 'a');
 
     if (Type.isObject(linkDetails) && Type.isString(linkDetails.href)) {
+      /**
+       * TINY-7993: Tiny's formatter sets an undesirable selection after applying a link.
+       * The caret is positioned right after the link, so `editor.selection.getNode()` returns a paragraph instead of the link.
+       * Consequently, the context menu and the link dialog do not work properly.
+       * We're using a bookmark here to restore the desirable selection.
+       */
+      const bookmark = editor.selection.getBookmark();
```

Comment:
This is definitely the wrong approach. If you look at the source JIRA comments, a simple selection collapse should suffice rather than using bookmarks - that's also the workaround used in the support case.

And depending on who you ask, customers inserting a link might not want the cursor to be in the link after closing the dialog (for example if they continue typing they probably don't want it to be inside the link).

Reply:
I found different approaches of how selection should behave after hyperlink insertion. For example, Jira moves collapsed selection after hyperlink, but CK editor lefts the entire hyperlink selected. It looks like TinyMCE choose the way similar to CK editor. So this PR tries to fix intended functionality but not to change it to the alternative approach.

Reply:
> This is definitely the wrong approach. If you look at the source JIRA comments, a simple selection collapse should suffice rather than using bookmarks - that's also the workaround used in the support case.

Unless I'm misunderstanding something, the function `applyLinkToSelection` operates only on the selection here. So bookmarking makes perfect sense here, no? 

> And depending on who you ask, customers inserting a link might not want the cursor to be in the link after closing the dialog (for example if they continue typing they probably don't want it to be inside the link).

As implied above, I believe that `applyLinkToSelection` works only within the selection. If a user inserts a link while writing text the link won't be selected, so one can proceed the writings. I've tested that manually and it's like so now.



> So this PR tries to fix intended functionality but not to change it to the alternative approach.

Viktor is 100% right here. We are not planning to change the current behavior, just fix the issue with the dialog and context menu.